packages/calcite-components/src/components/accordion-item/accordion-item.tsx

```diff
@@ -108,6 +111,11 @@ export class AccordionItem implements ConditionalSlotComponent {
   //  Lifecycle
   //
   //--------------------------------------------------------------------------
+  componentWillLoad(): void {
+    // TODO: check if this attribute is manually set by the user before getting the theme set in local storage.
+    // TODO: make this a part of the generic component setup.
+    this.calciteMode = getMode();
```

Comment:
This would likely need to happen on `connectedCallback()` so a component updates its mode if its moved position in the DOM.